openstates/de/bills.py

```diff
@@ -1,693 +1,329 @@
-        session_num = doc.xpath(xpath)[0].text_content()
-        session_num = re_digits.match(session_num).group()
-
-        #---------------------------------------------------------------------
-        # Sponsors
-        chamber = bill['chamber']
-
-        sponsor_types = {
-            'Additional Sponsor(s):': 'cosponsor',
-            'CoSponsors:': 'cosponsor',
-            'Primary Sponsor:': 'primary'}
-
-        xpath = '//font[contains(., "Sponsor") and @color="#008080"]'
-        headings = doc.xpath(xpath + '/text()')
-        sponsors = doc.xpath(xpath + '/../../following-sibling::td/font/text()')
-
-        for h, s in zip(headings, sponsors):
-
-            names = _cleanup_sponsors(s, chamber)
-            type_ = sponsor_types[h.strip()]
-
-            if names:
-                for name, _chamber in names:
-                    bill.add_sponsor(type_, name, chamber=_chamber)
-
-        #---------------------------------------------------------------------
-        # Versions
-
-        tmp = '/'.join([
-            'http://www.legis.delaware.gov',
-            'LIS/lis{session_num}.nsf/vwLegislation',
-            '{moniker}/$file/{filename}{format_}?open'])
-
-        documents = self.scrape_documents(source=url,
-                                     docname="introduced",
-                                     filename="Legis",
-                                     tmp=tmp,
-                                     session_num=session_num)
-
-        for d in documents:
-            bill.add_version(**d)
-
-        # If bill is a substitution, add the original as a version.
-        names = doc.xpath('//*[contains(text(), "Substituted '
-                          'Legislation for Bill:")]/text()')
-        urls = doc.xpath('//*[contains(text(), "Substituted '
-                          'Legislation for Bill:")]'
-                         '/following-sibling::a/@href')
-
-        for name, url in zip(names, urls):
-
-            name = re_substitution.match(name).group(1)
-            bill.add_version(name, url,
-                             description='original bill')
-
-        #---------------------------------------------------------------------
-        # Actions
-        actions = doc.xpath('//font[contains(., "Actions History")]'
-                            '/../following-sibling::table/descendant::td[2]')
-        actions = actions[0].text_content()
-        actions = filter(None, actions.splitlines())
-
-        for a in reversed(actions):
-            date, action = a.split(' - ', 1)
-            try:
-                date = datetime.strptime(date, '%b %d, %Y')
-            except ValueError:
-                date = datetime.strptime(date, '%B %d, %Y')  # XXX: ugh.
-
-            actor = actions_get_actor(action, bill['chamber'])
-            attrs = dict(actor=actor, action=action, date=date)
-            attrs.update(**self.categorizer.categorize(action))
-            bill.add_action(**attrs)
```

Comment:
Removed additional information from vote action (the counts are still in the text, since that's how we get it from the state, and the votes are still available as votes)